methods/guest/Cargo.toml

```diff
@@ -10,6 +10,7 @@ codegen-units = 1
 lto = "fat"
 
 [dependencies]
+alloy-sol-types = { version = "1.2.1" }
```

Comment:
We previously added the functions abi_encoded_size() and abi_encode() to the Checkpoint struct to simplify usage. I suggest doing the same with the new abi::Journal. Having this dependency would also simplify dependency management.
After doing so, we could probably remove those methods from the checkpoint, as they will no longer be needed.

Reply:
Added `encode` and `encode_sized` to journal.

This makes even more sense with the encoding having to be done both in the guest and also by the CLI input builder.